### PR TITLE
Add CMake support to Autotools + XCode/VisualC distributables + add avif to cmake config

### PR DESCRIPTION
**Pull request by madebr**
- Add avif to CMake's config file
- Create `sdl2_image-config.cmake` + `sdl2_Image-config-version.cmake` for the Xcode SDL2_image.framework distributable
For filling in the supported formats, I looked in `Xcode/SDL_image.xcodeproj/project.pbxproj`
- Create `sdl2_image-config.cmake` + `sdl2_Image-config-version.cmake` for the VisualC development distributable
For filling in the supported formats, I looked at `VisualC/SDL_image.vcxproj`
- Add `sdl2_image-config.cmake` + `sdl2_Image-config-version.cmake` for autotools.
This support is incomplete, as I didn't hook up all variables yet (I added fixme's).
I also didn't add both .cmake files to the source distribution, nor added them to the install target.
They should be installed in the following folder: `$(DESTDIR)$(libdir)/cmake/SDL2_image/`

Most of these changes are untested!

@slouken the autotools support is unfinished...
If you decide that the `SDL2_IMAGE_xxx` variables should not be exported, then the autotools/xcode/visualc cmake config files can be simplified.

### CODE REVIEW

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+set(SDL2IMAGE_JXL   FALSE)
+set(SDL2IMAGE_LBM   TRUE)
+set(SDL2IMAGE_PCX   TRUE)
+set(SDL2IMAGE_PNG   TRUE)
+set(SDL2IMAGE_PNM   TRUE)
+set(SDL2IMAGE_QOI   TRUE)
+set(SDL2IMAGE_SVG   TRUE)
+set(SDL2IMAGE_TGA   TRUE)
+set(SDL2IMAGE_TIF   TRUE)
+set(SDL2IMAGE_XCF   FALSE)
```
Comment (slouken) [inline Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: I believe this is TRUE

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+set(SDL2IMAGE_JPG   TRUE)
+set(SDL2IMAGE_JXL   FALSE)
+set(SDL2IMAGE_LBM   TRUE)
+set(SDL2IMAGE_PCX   TRUE)
+set(SDL2IMAGE_PNG   TRUE)
+set(SDL2IMAGE_PNM   TRUE)
+set(SDL2IMAGE_QOI   TRUE)
+set(SDL2IMAGE_SVG   TRUE)
+set(SDL2IMAGE_TGA   TRUE)
+set(SDL2IMAGE_TIF   TRUE)
```
Comment (slouken) [inline Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:21]: I believe this is FALSE

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 1 above)
Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: It's not in here:
https://github.com/libsdl-org/SDL_image/blob/01a2a971dd8b48e0b63dd809ee7aa1e899486ac6/Xcode/SDL_image.xcodeproj/project.pbxproj#L637-L652

Or am I looking in the wrong place?

Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: It's not in here:
https://github.com/libsdl-org/SDL_image/blob/01a2a971dd8b48e0b63dd809ee7aa1e899486ac6/Xcode/SDL_image.xcodeproj/project.pbxproj#L637-L652

Or am I looking in the wrong place?

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 2 above)
Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:21]: I got it from here:
https://github.com/libsdl-org/SDL_image/blob/01a2a971dd8b48e0b63dd809ee7aa1e899486ac6/Xcode/SDL_image.xcodeproj/project.pbxproj#L649

Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:21]: I got it from here:
https://github.com/libsdl-org/SDL_image/blob/01a2a971dd8b48e0b63dd809ee7aa1e899486ac6/Xcode/SDL_image.xcodeproj/project.pbxproj#L649

**Comment by slouken** [comment]
It seems reasonable to add the format variables to the script. That would allow projects that include SDL_image to verify that specific formats are supported, right? That seems helpful if you're including SDL_image from a binary package. I assume they would be used the same way when directly including the SDL_image directory in another project?

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 1 above)
Comment (slouken) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: Oh, sorry, I thought this was the Visual Studio support file. XCF support being missing is a bug. I'll fix that.

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 2 above)
Comment (slouken) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:21]: Oh, sorry, I thought this was the Visual Studio support file. That's correct.

**Comment by madebr** [comment]
Yes

Right now, they can do:
```cmake
find_package(SDL2_image 2.5 REQUIRED)
if(NOT SDL2IMAGE_AVIF)
    message(FATAL_ERROR "SDL2_image does not have avif support")
endif()
```

I didn't add it (yet), but with CMake component support, they can do the following:
e.g.
```cmake
# SDL2_image for the shared library, AVIF for libavif support
find_package(SDL2_image 2.5 REQUIRED COMPONENTS SDL2_image AVIF)
```

I didn't add variables for png/jpeg loading. Would these be useful?

**Comment by slouken** [comment]
> I didn't add variables for png/jpeg loading. Would these be useful?

Yes, every format should have an associated variable.

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 1 above)
Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: No worries, I didn't add anything to `Resources/ReadMe.txt` yet.
May I add the same text as I do in my pending SDL2 pr?

Comment (slouken) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake:22]: Yep, that sounds good. Also merge your SDL2 PR when you're ready.

File: configure.ac [hunk 3]
```diff
@@ -110,7 +111,16 @@ case "$host" in
...
-AM_CONDITIONAL(USE_IMAGEIO, test x$enable_imageio = xyes)
+case "$host" in
+    *-*-mingw*)
+        AC_SUBST([MINGW],1)
+        ;;
+    *)
+        AC_SUBST([MINGW],0)
+        ;;
+esac
+AM_CONDITIONAL(USE_IMAGEIO, $enable_imageio)
```
Comment (slouken) [inline configure.ac:122]: This needs to be the old conditional:
`AM_CONDITIONAL(USE_IMAGEIO, test x$enable_imageio = xyes)`

File: configure.ac [hunk 4]
```diff
@@ -110,7 +111,16 @@ case "$host" in
...
+AC_SUBST([USE_IMAGEIO], test x$enable_imageio = xyes)
```
Comment (slouken) [inline configure.ac:123]: This probably needs to be an actual variable that evaluates to 1 or 0. If you want you can comment this out for your commit and I can fix it up here.

File: sdl2_image-config.cmake.in [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# sdl2_image cmake project-config input for ./configure scripts
+
+set(SDL2_image_FOUND TRUE)
+
+set(SDL2IMAGE_AVIF  @LOAD_AVIF@)
```
Comment (slouken) [inline sdl2_image-config.cmake.in:5]: Can the substituted variables be 1 and 0, or do they need to be TRUE/FALSE, or ON/OFF?

Comment (madebr) [reply sdl2_image-config.cmake.in:5]: CMake does not care between true/false, TRUE/FALSE, on/off, or ON/OFF

https://cmake.org/cmake/help/latest/command/if.html#basic-expressions
> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers). False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. If the argument is not one of these specific constants, it is treated as a variable or string (see [Variable Expansion](https://cmake.org/cmake/help/latest/command/if.html#variable-expansion) further below) and one of the following two forms applies.

File: Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake (hunk 1 above)
Comment (madebr) [reply Xcode/pkg-support/resources/CMake/sdl2_image-config.cmake]: When enabling/disabling modules in VisualC and Xcode, don't forget to also update these CMake configuration files.

**Comment by madebr** [comment]
@slouken
Can you please review this pr and test the cmake integration into `SDL2_image.framework`?

**Comment by slouken** [comment]
Sure. I’m out of town for a few days, but I’ll check it out when I get back.

**Comment by slouken** [comment]
If you want, you can queue up changes for the other libraries so they’re ready to go when I get back.

**Comment by slouken** [comment]
Okay, I've merged and committed a few fixes for the packaging system, and verified that the CMake support files are present. It looks like you've added a test to verify it in the workflow?

**Pull request merged**